requests-for-comments/2022-11-08-supported-languages.md

```diff
@@ -0,0 +1,169 @@
+
+You may not be familiar with this one but it is _very_ safe and performant. Companies that have huge amounts of money on the line who are risk adverse to defects and slowdowns, like the HFT firm Jane Street use OCaml because it is unforgiving in how type safe it is.
+
+> OCaml’s powerful type system means more bugs are caught at compile time, and large, complex codebases are easier to maintain. This makes it a good language for running critical code. At the same time, sophisticated inference makes the type system unobtrusive, creating a smooth developer experience.
+
+The good:
+
+- Garbage collected
+- Algebraic data types
+- Pattern matching
+- Type inference
+- Immutable
+- Static Type-checking
+- First-class functions
+- Parametric polymorphism
+- Used by serious programming shops
+
+The bad:
+
+- Not terribly popular relative to the others
+- Can be considered relatively academic
+- No one experienced here
+
+### Elixir
+
+Oh, Erlang <3
+
+The good:
+
+- Built for streaming data
+- The cockroach of runtimes.
+- Automatic function level clustering
+- Hot reloading of functions
+- Compiled
+- Relatively functional
+
+The bad:
+
+- Dynamically typed
+- Runs in a VM
+
+### Scala / Java
+
+Grouping these together because they really are converging. Really this is any language on the JVM.
+
+The JVM is truly a wonderful runtime. It's very fast. The hotspot detection + Just In Time Compiling is _super_ impressive.
+
+The good:
+
+- Surprisingly performant
+- JVM Library interrop means huge availability of libraries out there
+- Tons of expertise and companies big and small are building software in this
+- Kafka and Zookeeper are built on this
+
+The bad:
+
+- Considered somewhat crusty
+- Slow boot times
+- JVM tuning is no fun
+
+## Discussion
+
+So with this overview laid out let's talk about our first scenario where we need to decide what language we should use:
+
+[RFC: Inserter Service Requirements](https://github.com/PostHog/meta/pull/68)
+
+The TL;DR requirements here are:
+
+- Consume from Kafka
+- Insert into ClickHouse
+- Deserialize some portion of payload (serialization TBD) to determine where to insert
+
+It's a simple service and could be written any anything really. It does give us an opportunity to branch off our well traveled path of Python and Typescript. In my opinion I think Golang or Rust would be a great fit here, as would the other languages listed. I'm a Gopher in particular so I would really like to see more written here.
+
+## Success criteria
+
+_How do we know if this is successful (i.e. metrics, customer feedback), what's out of scope, whats makes this ambitious?_
+
+The goal here is to spark debate about languages that we should use here at Posthog. What are we open to? Why should we not adopt new technologies.
+
+Success here would be to make a decision and effectively enact a policy on this and have engineers aligned and not worry about this again (at least for some time)
```

Comment:
As above... one decision only?

1) what needs to be done to make ingestion safer/faster/misc
2) how many programming languages can we support
3) how do we support any one programming language